test/hotspot/jtreg/compiler/vectorization/TestSubwordTruncation.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorization;
+
+import jdk.test.lib.Asserts;
+import compiler.lib.ir_framework.*;
+import compiler.lib.generators.*;
+
+/*
+ * @test
+ * @bug 8350177
+ * @summary Ensure that truncation of subword vectors produces correct results
+ * @library /test/lib /
+ * @run driver compiler.vectorization.TestSubwordTruncation
+ */
+
+public class TestSubwordTruncation {
+    private static final RestrictableGenerator<Integer> G = Generators.G.ints();
+    private static final int SIZE = 10000;
+
+    @Setup
+    static Object[] setupShortArray() {
+        short[] arr = new short[SIZE];
+        for (int i = 0; i < SIZE; i++) {
+            arr[i] = G.next().shortValue();
+        }
+
+        return new Object[] { arr };
+    }
+
+
+    @Setup
+    static Object[] setupByteArray() {
+        byte[] arr = new byte[SIZE];
+        for (int i = 0; i < SIZE; i++) {
+            arr[i] = G.next().byteValue();
+        }
+
+        return new Object[] { arr };
+    }
+
+    // Shorts
+
```

Comment:
```suggestion
```
Nit: you don't have a similar comment for other types, so just drop it here too ;)

Reply:
Further in the file I used `// Bytes` to separate the byte methods, as well as the new char methods I added in the new commit. I think it helps navigating the file at a glance, at least in my emacs editor :)